java-checks/src/test/resources/checks/SingleConnectionFactoryCheck/beans.xml

```diff
@@ -26,4 +26,15 @@
 
   <bean id="scf5" class="org.springframework.jms.connection.SingleConnectionFactory" p:reconnectOnException="false"> <!-- Noncompliant {{Add a "reconnectOnException" property, set to "true"}} -->
   </bean>
+
+  <bean id="scf6" class="org.springframework.jms.connection.SingleConnectionFactory"> <!-- Compliant -->
+    <property name="reconnectOnException" value="true">
```

Comment:
The fix you wrote probably supports code such as: `<property name="reconnectOnException" value="false" />`, we should add a test to confirm this.

Reply:
I have take your comment see second commit